gcc/rust/resolve/rust-ast-resolve-pattern.h

```diff
@@ -55,45 +63,25 @@ class PatternDeclaration : public ResolverBase
   using Rust::Resolver::ResolverBase::visit;
 
 public:
-  static void go (AST::Pattern *pattern, Rib::ItemType type)
-  {
-    PatternDeclaration resolver (type);
-    pattern->accept_vis (resolver);
-  };
-
-  void visit (AST::IdentifierPattern &pattern) override
-  {
-    // if we have a duplicate id this then allows for shadowing correctly
-    // as new refs to this decl will match back here so it is ok to overwrite
-    resolver->get_name_scope ().insert (
-      CanonicalPath::new_seg (pattern.get_node_id (), pattern.get_ident ()),
-      pattern.get_node_id (), pattern.get_locus (), type);
-  }
+  static void go (AST::Pattern *pattern, Rib::ItemType type);
+  static void go (AST::Pattern *pattern, std::vector<PatternBinding> *bindings,
```

Comment:
If your adding another override you should keep the pattern the same so you have:

```c++
static void go (AST::Pattern *pattern, Rib::ItemType type);
static void go (AST::Pattern *pattern, Rib::ItemType type, std::vector<PatternBinding>* bindings);
```